get-python-version/action.yml

```diff
@@ -26,6 +26,9 @@ runs:
     run: |
       if ${{ inputs.micro }} ; \
       then export version=$(sed -n 's/^PYTHON_VERSION = \([0-9]\.[0-9]\{1,2\}\.[0-9]\{1,2\}\).*/\1/p' Makefile 2>1 || true); \
+      if [[ ! $version ]]; \
+      then export version=$(sed -n 's/^PYTHON_VERSION = \([0-9]\.[0-9]\{1,2\}\).*/\1/p' Makefile 2>1 || true); \
+      fi
       else export version=$(sed -n 's/^PYTHON_VERSION = \([0-9]\.[0-9]\{1,2\}\).*/\1/p' Makefile 2>1 || true); \
       fi
```

Comment:
I think the else is no longer necessary, right? Either you succeed extracting version with micro when you require micro. And when you fail, or don't require micro, version is undefined, so the ```if``` suffices.

Reply:
I want to log that the micro version wasn't found when it was desired.

I've refactored a bit so that a) we now have indentation - yay and b) the extraction without micro is run only once. I don't see how to reduce this further without losing some info.

Reply:
Sounds good.